imap-uw failed to build on macOS 13.5

### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
https://gist.github.com/crazywhalecc/e32b19f0f1cbc9f2e3114cc1799f024c
```


### Verification

- [X] My "`brew doctor` output" says `Your system is ready to brew.` and am still able to reproduce my issue.
- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.
- [X] I searched for recent similar issues at https://github.com/Homebrew/homebrew-core/issues?q=is%3Aissue and found no duplicates.

### What were you trying to do (and why)?

I want to compile and build imap-uw locally, found some errors. And I remembered that I could refer to homebrew to compile the imap-uw code, so I tried building it from source using homebrew first.

### What happened (include all command output)?

```
==> Fetching imap-uw
==> Downloading https://raw.githubusercontent.com/Homebrew/homebrew-core/2e3073c4e90275a30d6d7220f62eaa40ebdbf5ec/Formula/i/imap-uw.rb
Already downloaded: /Users/jerry/Library/Caches/Homebrew/downloads/fa4d785233cf007af84d3d33da361bd924ebf430893a564c281a773a94fb2748--imap-uw.rb
==> Downloading https://salsa.debian.org/holmgren/uw-imap/raw/master/debian/patches/1006_openssl1.1_autoverify.patch
######################################################################################################################################### 100.0%
==> Downloading https://salsa.debian.org/holmgren/uw-imap/raw/master/debian/patches/2014_openssl1.1.1_sni.patch
######################################################################################################################################### 100.0%
==> Downloading https://mirrorservice.org/sites/ftp.cac.washington.edu/imap/imap-2007f.tar.gz
Already downloaded: /Users/jerry/Library/Caches/Homebrew/downloads/96d819744d8c1c612ef05e67712c9a40f4433cd32c869c8d0804edfdf4516230--imap-2007f.tar.gz
==> Patching
==> Applying 1006_openssl1.1_autoverify.patch
patching file 'src/osdep/unix/ssl_unix.c'
==> Applying 2014_openssl1.1.1_sni.patch
patching file 'src/osdep/unix/ssl_unix.c'
==> make oxp
Last 15 lines from /Users/jerry/Library/Logs/Homebrew/imap-uw/01.make:
`cat CCTYPE` -c `cat CFLAGS` utf8.c
`cat CCTYPE` -c `cat CFLAGS` utf8aux.c
`cat CCTYPE` -c `cat CFLAGS` siglocal.c
`cat CCTYPE` -c `cat CFLAGS` dummy.c
dummy.c:617:28: error: call to undeclared function 'time'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
    stream->uid_validity = time (0);
                           ^
dummy.c:642:7: error: call to undeclared function 'time'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
  if (time (0) >=               /* time to do another test? */
      ^
2 errors generated.
make[3]: *** [dummy.o] Error 1
make[2]: *** [osx] Error 2
make[1]: *** [OSTYPE] Error 2
make: *** [oxp] Error 2

READ THIS: https://docs.brew.sh/Troubleshooting

```

### What did you expect to happen?

Build imap-uw successfully.

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
brew install imap-uw --build-from-source
```

--- a/Formula/i/imap-uw.rb
+++ b/Formula/i/imap-uw.rb
@@ -60,12 +60,16 @@ class ImapUw < Formula
     # Skip IPv6 warning on Linux as libc should be IPv6 safe.
     touch "ip6"
 
+    args = []
+    # Workaround for Xcode 14.3.
+    args << "EXTRACFLAGS=-Wno-implicit-function-declaration" if DevelopmentTools.clang_build_version >= 1403
+
     target = if OS.mac?
       "oxp"
     else
       "ldb"
     end
-    system "make", target
+    system "make", target, *args
 
     # email servers:
     sbin.install "imapd/imapd", "ipopd/ipop2d", "ipopd/ipop3d"

